tests/queries/0_stateless/01455_leaf_max_rows_bytes_to_read.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "$CURDIR"/../shell_config.sh
+
+$CLICKHOUSE_CLIENT --query="DROP TABLE IF EXISTS test_local;"
+$CLICKHOUSE_CLIENT --query="DROP TABLE IF EXISTS test_distributed;"
+
+$CLICKHOUSE_CLIENT --query="CREATE TABLE test_local (date Date, value UInt32) ENGINE = MergeTree(date, date, 8192);"
+$CLICKHOUSE_CLIENT --query="CREATE TABLE test_distributed AS test_local ENGINE = Distributed(test_shard_localhost, currentDatabase(), test_local, rand());"
+$CLICKHOUSE_CLIENT --query="INSERT INTO test_local SELECT '2000-08-01', number as value from numbers(50000);"
+
+# Must throw an exception
+EXCEPTION_TEXT="Limit for rows (controlled by 'max_rows_to_read_leaf' setting) exceeded"
+$CLICKHOUSE_CLIENT --query="SELECT count() FROM (SELECT * FROM test_distributed) SETTINGS max_rows_to_read_leaf = 40000;" 2>&1 \
+    | grep -q "$EXCEPTION_TEXT" && echo "$EXCEPTION_SUCCESS_TEXT" || echo "Did not thrown an exception"
+
+# Must throw an exception
```

Comment:
Do you know that we have annotations for expected server or client error in SQL tests? It's written like this:
```
select x; -- { serverError 49 }
```
This test expects error about unknown column 'x', code 49. Looks like you might be able to rewrite this test as SQL using these annotations, which is preferrable to bash for many reasons (e.g. it's easier to do query fuzzing on SQL tests).

Reply:
Most of these tests in this `.sh` file can be written via a `.sql` file. Like @akuzm mentioned, for the queries that you expect to error, you can annotate the respective error code next to the line as comment. Error codes are either client `{clientError xx}` or server errors and you can usually see those codes from the exceptions that the clickhouse-client throws).

One added advantage of using `.sql` file is for queries like `DROP TABLE IF EXISTS test_local`  you don't have to leave blank lines in the `.reference` like you currently have.